standards/managing-services-interactions.md

```diff
@@ -0,0 +1,45 @@
+---
+category: Building software
+expires: 2018-03-29
+---
+# Managing interactions between services
+
+Services need to interact with one another over APIs. They need to be
+able to find one another, authenticate clients, be fault-tolerant,
+scale appropriately in line with demand, and more.
+
+## Principles
+
+Services should be built on the assumption that [whatever network they
+reside in is
+insecure](https://en.wikipedia.org/wiki/Fallacies_of_distributed_computing).
+It should be possible to deploy a new instance of a service to a new
+environment (with appropriate configuration) and have it present an
+appropriately-secured interface to the network.
+
+A failure (including high latency) in a downstream service should not
+inherently mean a failure in a consuming service, but it is not possible
+for a third party service to accurately decide the appropriate response
+in the case of a failure. Only the provider or consumer of the API can
+have enough knowledge of failure states to handle them.
+
+Handling increased load is best done at the level of the service.
```

Comment:
Which service, the consumer or the provider?

Reply:
A good point. I guess it's actually both? The consumer needs to handle “no, not now” responses sensibly, and the provider needs to serve those at the right time, too.